### PR TITLE
Allow register_service without a class name

### PR DESCRIPTION
**Pull request by FrankBakkerNl**
## Proposed change

Allow register_service without a class name so we can use it from the new model

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionalit)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass.
- [ ] There is no commented out code in this PR.
- [ ] The code has been formatted using Black (`make black`)
- [ ] Tests have been added to verify that the new code works.

### CODE REVIEW

File: custom_components/netdaemon/__init__.py [hunk 1]
```diff
@@ -60,12 +60,16 @@ async def async_update_data():
...
+        if daemon_class:
+            service_name = f"{daemon_class}_{daemon_method}"
+        else:
+            service_name = daemon_method
 
-        LOGGER.info("Register service %s_%s", daemon_class, daemon_method)
+        LOGGER.info("Register service %s", service_name)
         hass.services.async_register(
-            DOMAIN, f"{daemon_class}_{daemon_method}", netdaemon_noop
+            DOMAIN, service_name, netdaemon_noop
```
Comment (ludeeus) [inline custom_components/netdaemon/__init__.py:72]: This is now a breaking change, why exactly is that needed?

Comment (helto4real) [reply custom_components/netdaemon/__init__.py:72]: yea it is a breaking change. maybe we should add a new method for this instead so we do not break existing running ND versions?

Comment (ludeeus) [reply custom_components/netdaemon/__init__.py:72]: Breaking changes are fine if there is a good reason for them

Comment (FrankBakkerNl) [reply custom_components/netdaemon/__init__.py:72]: I think it is not breaking, only when class was empty we now ignore it and also do not add the _

Comment (ludeeus) [reply custom_components/netdaemon/__init__.py:72]: It's definitely breaking

Comment (FrankBakkerNl) [reply custom_components/netdaemon/__init__.py:72]: to explain, from ND the old way we use this is always send the name of the class and the method to the integration and that merges them into the name of the new service. 

In the new api the callback is not a class with a method, but just a callback delegate. This means the user should provide the service name himself and it does not make sense to have him specify two parts that get merged with an _. To make this change simple I just used the method parameter and left the class empty. The only way this is breaking is when the user called this service directly (which is not the usual scenario) AND left the class empty end expected the service to be called "no class provided_...". An alternative would be to have a third argument 'service_name' and when that is filled the integration uses that and ignores class and method

**Review by ludeeus — APPROVED** [review]

**Pull request closed without merging**